### DISCUSSION TITLE
Measuring the execution time of .NET methods within Simple/Transaction controllers

### DISCUSSION THREAD

**Discussion by nmelnic** [abstracta/jmeter-dotnet-dsl#25] [Q&A]
Hi, dear community,

We are looking to use a performance testing library within the .NET ecosystem to measure the performance/response time of non-HTTP methods. Specifically, we want to call some .NET-based methods to replicate a flow executed in a desktop app within thread group(s), and then take advantage of other available features, such as reports, think times, etc.

Can this be achieved with JMeter DSL for .NET somehow, maybe extending Base Controller, or calling the method within Transaction Controller?

**Comment by rabelenda** [comment]
Hello, excellent question, thank you!

We are in the process of implementing proper support for lambdas in .Net. This would open a new set of possibilities, like solving some of the requirements that have been proposed in [this issue](https://github.com/abstracta/jmeter-dotnet-dsl/issues/3).

We still need to do some work to complete it, and properly evaluate it's performance. It would allow to execute any .Net code as part of the execution and cover the scenario that you are considering.

Until that is not ready the only option I can think of is packaging your .Net code as an executable application (or a rest service) and invoke such application (or api) from jmeter code via groovy invoking the exec app (or using http requests if you expose it as a rest service).

**Reply by nmelnic** [reply]
Many thanks for your quick reply.

I really appreciate this initiative, and there aren't that many libraries for writing performance tests in the .NET ecosystem. I used the DLS for Java (most of the available capabilities, even integration with Influx) and enjoyed it a lot.

By the way, is there any ETA for releasing the capability described above?

**Reply by rabelenda** [reply]
We have no ETA, but I hope is something we can complete before the end of the year (it might be in a pair of weeks, or a pair of months, depending on how other things move). If you have some need this ASAP for implementation in your company, and want to speed this up, you can try the enterprise support.
